rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3407,6 +4105,139 @@ int LuaUnsyncedCtrl::SetWindowMaximized(lua_State* L)
+ * @string startScript
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::Restart(lua_State* L)
+{
+	// same as Reload now, cl-args are always ignored
+	return (ReloadOrRestart(luaL_checkstring(L, 1), luaL_checkstring(L, 2), false));
+}
+
+
+/*** Launches a new Spring instance without terminating the existing one.
+ *
+ * @function Spring.Start
+ *
+ * If this call returns, something went wrong
+ *
+ * @string commandline_args commandline arguments passed to spring executable.
+ * @string startScript the CONTENT of the script.txt spring should use to start (if empty, no start-script is added, you can still point spring to your custom script.txt when you add the file-path to commandline_args.
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::Start(lua_State* L)
+{
+	if (ReloadOrRestart(luaL_checkstring(L, 1), luaL_checkstring(L, 2), true) != 0) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	return 0;
+}
+
+
+/*** Sets the icon for the process which is seen in the OS task-bar and other places (default: spring-logo).
+ *
+ * @function Spring.SetWMIcon
+ *
+ * Note: has to be 24bit or 32bit.
+ * Note: on windows, it has to be 32x32 pixels in size (recommended for cross-platform)
+ * Note: *.bmp images have to be in BGR format (default for m$ ones).
+ * Note: *.ico images are not supported.
+ *
+ * @string iconFileName
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::SetWMIcon(lua_State* L)
+{
+	const std::string iconFileName = luaL_checksstring(L, 1);
+	const bool forceResolution = luaL_optboolean(L, 2, false);
+
+	CBitmap iconTexture;
+
+	if (iconTexture.Load(iconFileName)) {
+		WindowManagerHelper::SetIcon(&iconTexture, forceResolution);
+	} else {
+		luaL_error(L, "Failed to load image from file \"%s\"", iconFileName.c_str());
+	}
+
+	return 0;
+}
+
+
+/*** Sets the window title for the process (default: "Spring <version>").
+ *
+ * @function SetWMCaption
+ *
+ * The shortTitle is displayed in the OS task-bar (default: "Spring <version>").
+ *
+ * NOTE: shortTitle is only ever possibly used under X11 (Linux & OS X), but not with QT (KDE) and never under Windows.
+ *
+ * @string title
+ * @string[opt=title] titleShort
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::SetWMCaption(lua_State* L)
+{
+	globalRendering->SetWindowTitle(luaL_checksstring(L, 1));
+	return 0;
+}
+
+
+/*** @function Spring.ClearWatchDogTimer
+ * @string[opt=main] threadName
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::ClearWatchDogTimer(lua_State* L) {
+	if (lua_gettop(L) == 0) {
+		// clear for current thread
+		Watchdog::ClearTimer();
+		return 0;
+	}
+
+	if (lua_isstring(L, 1)) {
+		Watchdog::ClearTimer(lua_tostring(L, 1), luaL_optboolean(L, 2, false));
+	} else {
+		Watchdog::ClearTimer("main", luaL_optboolean(L, 2, false));
+	}
+
+	return 0;
+}
+
+
+/*** @function Spring.SetClipboard
+ * @string text
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::SetClipboard(lua_State* L)
+{
+	SDL_SetClipboardText(luaL_checkstring(L, 1));
+	return 0;
+}
+
+
```

Comment:
/*** Relinquish control of the game loading thread and OpenGL context back to the UI (LuaIntro) to draw screen updates and process windows events. Should be called after each widget/unsynced gadget is loaded in widget/gadget handler.
 *
 * @function Spring.Yield
 *
 * if this call returns true then caller should continue calling Spring.Yield during the widgets/gadgets load, if false - it shouldn't call it any longer.
 *
 * @number sleep time in milliseconds.
 * @treturn bool
 */

Reply:
```cpp
/*** Relinquish control of the game loading thread and OpenGL context back to the UI (LuaIntro) to draw screen updates and process windows events.
  *
   * @function Spring.Yield
   *  Should be called after each widget/unsynced gadget is loaded in widget/gadget handler.
   * @number sleep time in milliseconds.
   * @treturn bool if this call returns true then caller should continue calling Spring.Yield during the widgets/gadgets load, if false - it shouldn't call it any longer.
    */
```